cirq-google/cirq_google/engine/abstract_local_processor.py

```diff
@@ -64,30 +74,30 @@ def __init__(
         engine: Optional['AbstractEngine'] = None,
         expected_down_time: Optional[datetime.datetime] = None,
         expected_recovery_time: Optional[datetime.datetime] = None,
-        schedule: Optional[List[qtypes.QuantumTimeSlot]] = None,
+        schedule: Optional[List[quantum.QuantumTimeSlot]] = None,
         project_name: str = 'fake_project',
     ):
         self._engine = engine
         self._expected_recovery_time = expected_recovery_time
         self._expected_down_time = expected_down_time
-        self._reservations: Dict[str, qtypes.QuantumReservation] = {}
+        self._reservations: Dict[str, quantum.QuantumReservation] = {}
         self._resource_id_counter = 0
         self._processor_id = processor_id
         self._project_name = project_name
 
         if schedule is None:
             self._schedule = [
-                qtypes.QuantumTimeSlot(
+                quantum.QuantumTimeSlot(
                     processor_name=self._processor_id,
-                    slot_type=qenums.QuantumTimeSlot.TimeSlotType.UNALLOCATED,
+                    time_slot_type=quantum.QuantumTimeSlot.TimeSlotType.UNALLOCATED,
                 )
             ]
         else:
             self._schedule = copy.copy(schedule)
-            self._schedule.sort(key=lambda t: t.start_time.seconds or -1)
+            self._schedule.sort(key=lambda t: t.start_time.timestamp() if t.start_time else -1)
```

Comment:
Yes, with `proto-plus` accessing the `start_time` field now gives a `datetime` rather than a `Timestamp`. It would be possible to compare actual datetimes here, but it's more awkward to construct the default value because it has to have a timezone or the comparison will fail. Comparing by timestamp seems simpler.

Reply:
Your response further threw me because a `datetime.timestamp()` is not a `Timestamp` lol. All good now, thanks for the explanation!